src/module.c

```diff
@@ -12856,37 +12856,59 @@ long long getModuleNumericConfig(ModuleConfig *module_config) {
     return module_config->get_fn.get_numeric(rname, module_config->privdata);
 }
 
+int loadModuleSingleConfig(dictEntry *config_queue_entry, ModuleConfig *module_config, bool set_default_if_missing) {
+    const char *err = NULL;
+    if (config_queue_entry) {
+        if (!performModuleConfigSetFromName(dictGetKey(config_queue_entry), dictGetVal(config_queue_entry), &err)) {
+            serverLog(LL_WARNING, "Issue during loading of configuration %s : %s", (sds) dictGetKey(config_queue_entry), err);
+            dictFreeUnlinkedEntry(server.module_configs_queue, config_queue_entry);
+            dictEmpty(server.module_configs_queue, NULL);
```

Comment:
i don't like the fact the dict iterator is in another function and this one is the one that does the unlinking.
i think i'd rather move this back into loadModuleConfigs, and have some code duplication in calling performModuleConfigSetDefaultFromName from two places.